.circleci/config.yml

```diff
@@ -1,57 +1,78 @@
 version: 2.1
 
 orbs:
-  docker: circleci/docker@2.2.0
+  docker: circleci/docker@2.8.0
 
+commands:
+  setup-buildkit-builder:
+    steps:
+      - run:
+          name: Setup BuildKit builder
+          command: |
+            docker buildx create \
+            --name zstd-builder \
+            --use \
+            --driver docker-container \
+            --driver-opt image=moby/buildkit:v0.17.0
 jobs:
   build-and-push-python:
-    executor: docker/docker
+    machine:
+      image: ubuntu-2204:current
+      docker_layer_caching: true
     parameters:
       python-version:
         type: string
     environment:
       PYTHON_VERSION: << parameters.python-version >>
     steps:
       - checkout
-      - setup_remote_docker
+      - setup-buildkit-builder
       - docker/check:
           use-docker-credentials-store: true
       - docker/build:
           step-name: Base Python image
           path: ./python/base
           dockerfile: Dockerfile
-          extra_build_args: "--build-arg FROM_PYTHON_VERSION=${PYTHON_VERSION}"
-          image: deepnote/python
-          tag: ${PYTHON_VERSION}${CIRCLE_PULL_REQUEST:+-ra-${CIRCLE_PULL_REQUEST##*/}}
-      - docker/push:
+          extra_build_args: >-
+            --build-arg FROM_PYTHON_VERSION=${PYTHON_VERSION}
+            --output='type=image,push=true,compression=zstd,force-compression=true,oci-mediatypes=true'
+            --platform linux/amd64
+            --progress plain
+          use-buildkit: true
           image: deepnote/python
           tag: ${PYTHON_VERSION}${CIRCLE_PULL_REQUEST:+-ra-${CIRCLE_PULL_REQUEST##*/}}
+
   build-and-push-python-datascience:
-    executor: docker/docker
+    machine:
+      image: ubuntu-2204:current
+      docker_layer_caching: true
     parameters:
       python-version:
         type: string
     environment:
       PYTHON_VERSION: << parameters.python-version >>
     steps:
       - checkout
-      - setup_remote_docker
+      - setup-buildkit-builder
       - docker/check:
           use-docker-credentials-store: true
       - docker/build:
           step-name: Datascience Python image
           path: ./python/datascience
           docker-context: ./python/datascience
           dockerfile: Dockerfile.datascience
-          extra_build_args: "--build-arg FROM_PYTHON_VERSION=${PYTHON_VERSION}"
-          image: deepnote/python-datascience
-          tag: ${PYTHON_VERSION}${CIRCLE_PULL_REQUEST:+-ra-${CIRCLE_PULL_REQUEST##*/}}
-      - docker/push:
+          extra_build_args: >-
+            --build-arg FROM_PYTHON_VERSION=${PYTHON_VERSION}
+            --output='type=image,push=true,compression=zstd,force-compression=true,oci-mediatypes=true'
```

Comment:
Is this way of building an image compatible when someone uses it as the basis for another image?